fields requirements changes

kevinkhao opened the pull request:
[IMP] Modifications related to changing "required" attribute on datamodels: includes SO generation, tests, components, demo data

Add tests for cases where data content is full/minimal/mixed
Fix state_code: search also according to country_id
Fix state_code, country_code validation and search

kevinkhao committed:
[IMP] Modifications related to changing "required" attribute on datamodels: includes SO generation, tests, components, demo data
Add tests for cases where data content is full/minimal/mixed
Fix state_code: search also according to country_id
Fix state_code, country_code validation and search

kevinkhao committed:
Merge branch '12.0' into 12.0-requirement_changes

kevinkhao committed:
[IMP] Add date to sale order

kevinkhao committed:
[FIX] Revert required attr on amounts

sebastienbeau commented on sale_import_base/components/importer_sale_channel.py:
```diff
@@ -50,17 +50,21 @@ def _validate_address(self, address, all_errors):
-                [("code", "=", address["state_code"])]
+                [
+                    ("code", "=", address["state_code"]),
+                    ("country_id", "in", country and country.ids),
```
country variable can be not defined

kevinkhao replied:
fixed

sebastienbeau commented on sale_import_base/components/importer_sale_channel.py:
```diff
@@ -165,16 +169,19 @@ def _prepare_partner(self, data):
                 [("code", "=", data["country_code"])]
             )
             result["country_id"] = country.id
+        if data.get("state_code"):
```
duplicated code

sebastienbeau commented on sale_import_base/datamodels/sale_order_amount.py:
```diff
@@ -7,6 +7,6 @@
-    amount_tax = fields.Decimal()
-    amount_untaxed = fields.Decimal()
-    amount_total = fields.Decimal()
+    amount_tax = fields.Decimal(required=True)
```
why?

sebastienbeau replied:
should be not required

kevinkhao replied:
sorry, commit in local was not yet pushed

kevinkhao committed:
Misc. feedback improvements

kevinkhao committed:
Merge branch '12.0-requirement_changes' of github.com:akretion/sale-import into 12.0-requirement_changes

kevinkhao committed:
make travis happy

kevinkhao committed:
[REF] On import, remove pre-emptive validation and error list, instead raise on first error
[IMP] Validate payment currency as it should match Sale Order's pricelist's currency

sebastienbeau commented on sale_import_base/components/importer_sale_channel.py:
```diff
@@ -165,16 +107,27 @@ def _prepare_partner(self, data):
+            state = self.env["res.country.state"].search(
+                [
+                    ("code", "=", data["state_code"]),
+                    ("country_id", "in", [result.get("country_id")]),
```
("country_id", "=", result.get("country_id")),

kevinkhao replied:
this whole part is made conditional on there being a country

kevinkhao committed:
[IMP] More sensible logic on state according to feedback
